### PR TITLE
Grammar keyword fix

### LINKED ISSUE #4328: Scale keyword in UDT as a member name is a parser error
Stumbled upon this while I was playing with a `tagDEC` struct:

```
Option Explicit

' https://docs.microsoft.com/en-us/windows/desktop/api/wtypes/ns-wtypes-tagdec
Private Type tagDec
    wReserved As Integer
    scale As Byte
    sign As Byte
    Hi32 As Long
    Lo64Lo As Long
    Lo64Hi As Long
End Type
```
The parser is choking on the `scale` member because it's a keyword.

Relevant log bit:

```
2018-08-30 11:16:07.1279;ERROR-2.2.0.3624;Rubberduck.Parsing.VBA.Parsing.ModuleParser;Syntax error; offending token 'scale' at line 6, column 5 in the CodePaneCode version of module Module1.;
2018-08-30 11:16:07.1279;DEBUG-2.2.0.3624;Rubberduck.Parsing.VBA.Parsing.ModuleParser;SyntaxErrorException thrown in thread 31, ParseTaskID a1d3ada6-85ce-4f61-98a9-30829356b3bf.;Rubberduck.Parsing.Symbols.ParsingExceptions.MainParseSyntaxErrorException: mismatched input 'scale' expecting END_TYPE ---> Antlr4.Runtime.InputMismatchException: Exception of type 'Antlr4.Runtime.InputMismatchException' was thrown.
   at Antlr4.Runtime.DefaultErrorStrategy.RecoverInline(Parser recognizer)
   at Antlr4.Runtime.Parser.Match(Int32 ttype)
   at Rubberduck.Parsing.Grammar.VBAParser.udtDeclaration() in C:\projects\rubberduck\Rubberduck.Parsing\obj\Release\VBAParser.cs:line 13232
   --- End of inner exception stack trace ---
   at Rubberduck.Parsing.Symbols.ParsingExceptions.MainParseExceptionErrorListener.SyntaxError(IRecognizer recognizer, IToken offendingSymbol, Int32 line, Int32 charPositionInLine, String msg, RecognitionException e) in C:\projects\rubberduck\Rubberduck.Parsing\Symbols\ParsingExceptions\MainParseExceptionErrorListener.cs:line 17
   at Antlr4.Runtime.ProxyErrorListener`1.SyntaxError(IRecognizer recognizer, Symbol offendingSymbol, Int32 line, Int32 charPositionInLine, String msg, RecognitionException e)
   at Antlr4.Runtime.Parser.NotifyErrorListeners(IToken offendingToken, String msg, RecognitionException e)
   at Antlr4.Runtime.DefaultErrorStrategy.NotifyErrorListeners(Parser recognizer, String message, RecognitionException e)
   at Antlr4.Runtime.DefaultErrorStrategy.ReportInputMismatch(Parser recognizer, InputMismatchException e)
   at Antlr4.Runtime.DefaultErrorStrategy.ReportError(Parser recognizer, RecognitionException e)
   at Rubberduck.Parsing.Grammar.VBAParser.udtDeclaration() in C:\projects\rubberduck\Rubberduck.Parsing\obj\Release\VBAParser.cs:line 13237
   at Rubberduck.Parsing.Grammar.VBAParser.moduleDeclarationsElement() in C:\projects\rubberduck\Rubberduck.Parsing\obj\Release\VBAParser.cs:line 2257
   at Rubberduck.Parsing.Grammar.VBAParser.moduleDeclarations() in C:\projects\rubberduck\Rubberduck.Parsing\obj\Release\VBAParser.cs:line 1945
   at Rubberduck.Parsing.Grammar.VBAParser.module() in C:\projects\rubberduck\Rubberduck.Parsing\obj\Release\VBAParser.cs:line 442
   at Rubberduck.Parsing.Grammar.VBAParser.startRule() in C:\projects\rubberduck\Rubberduck.Parsing\obj\Release\VBAParser.cs:line 332
   at Rubberduck.Parsing.VBA.Parsing.VBATokenStreamParser.Parse(ITokenStream tokenStream, PredictionMode predictionMode, IParserErrorListener errorListener) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\VBATokenStreamParser.cs:line 21
   at Rubberduck.Parsing.VBA.Parsing.TokenStreamParserBase.ParseSll(String moduleName, ITokenStream tokenStream, CodeKind codeKind) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\TokenStreamParserBase.cs:line 84
   at Rubberduck.Parsing.VBA.Parsing.TokenStreamParserBase.ParseWithFallBack(String moduleName, CommonTokenStream tokenStream, CodeKind codeKind) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\TokenStreamParserBase.cs:line 52
   at Rubberduck.Parsing.VBA.Parsing.TokenStreamParserBase.Parse(String moduleName, CommonTokenStream tokenStream, CodeKind codeKind, ParserMode parserMode) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\TokenStreamParserBase.cs:line 32
   at Rubberduck.Parsing.VBA.Parsing.TokenStreamParserStringParserAdapterWithPreprocessing.Parse(String moduleName, String projectId, String code, CancellationToken token, CodeKind codeKind, ParserMode parserMode) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\TokenStreamParserStringParserAdapterWithPreprocessing.cs:line 29
   at Rubberduck.Parsing.VBA.Parsing.ModuleParser.CodePanePassResults(QualifiedModuleName module, CancellationToken token, TokenStreamRewriter rewriter) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\ModuleParser.cs:line 167
   at Rubberduck.Parsing.VBA.Parsing.ModuleParser.ParseInternal(QualifiedModuleName module, CancellationToken cancellationToken, TokenStreamRewriter rewriter, Guid taskId) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\ModuleParser.cs:line 87
   at Rubberduck.Parsing.VBA.Parsing.ModuleParser.Parse(QualifiedModuleName module, CancellationToken cancellationToken, TokenStreamRewriter rewriter) in C:\projects\rubberduck\Rubberduck.Parsing\VBA\Parsing\ModuleParser.cs:line 43
Token: scale at L6C5
Kind of parsed code: CodePaneCode
Component: Module1 (code pane version)
ParseType: Main parse
```

### PR DESCRIPTION
**Pull request by MDoerner**
This PR closes #4328 .

The special form keywords `PSET`, `SCALE` and `CIRCLE` were added to the `statementKeyword` parser rule at the time the tokens were added to the lexer rules. This caused the issue closed with this PR.

Note however that this breaks the SLL parser for the `circleSpecialForm` and `psetSpecialForm` in two veru edge cases:

1. The special form is used on a variable, the `STEP` is not omitted, but the whitespace following it is, e.g. 
```vba
Me.Circle Step(1,2), 3
```
2. The special form is used on a variable, the `STEP` is not omittedand the whitespace following it contains a line continuateion, e.g. 
```vba
Me.Circle Step _
(1,2), 3
```

The problem here is that both could legally be expressions were it not for `Circle` being a `statementKeyword`, a distinction very hard to incorporate into the grammar. (Step is interpreted as a function here.) This makes the SLL parser fail, because it is context free, i.e. while matching the `expression` subrule of the special form, it does not know that a `DOT` has to come next, followed by `CIRCLE` or `PSET`.  

Because these two are very far to the edge, I adjusted two failing tests for parsing the `circleSpecialForm` to have ordinary whitespace after `STEP`.

### CODE REVIEW

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 1]
```diff
@@ -161,10 +161,10 @@ mainBlockStmt :
     | variableStmt
     | whileWendStmt
     | withStmt
-    | lineSpecialForm
     | circleSpecialForm
     | scaleSpecialForm
     | pSetSpecialForm
+    | lineSpecialForm
```
Comment (Vogel612) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: Not sure why this reordering was done. Can you shed some light?

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 2]
```diff
@@ -567,7 +567,7 @@ withStmt :
 
 // Special forms with special syntax, only available in VBA reports or VB6 forms and pictureboxes.
 lineSpecialForm : expression whiteSpace ((STEP whiteSpace?)? tuple)? MINUS (STEP whiteSpace?)? tuple whiteSpace? (COMMA whiteSpace? expression)? whiteSpace? (COMMA whiteSpace? lineSpecialFormOption)?;
-circleSpecialForm : (expression whiteSpace? DOT whiteSpace?)? CIRCLE whiteSpace (STEP whiteSpace?)? tuple (whiteSpace? COMMA whiteSpace? expression)+;
+circleSpecialForm : (expression whiteSpace? DOT whiteSpace?)? CIRCLE (whiteSpace STEP)? whiteSpace? tuple (whiteSpace? COMMA whiteSpace? expression)+;
```
Comment (Vogel612) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: As it stands, (and IIUC) the new rule allows for `Circle(1,3) 2`, note the missing space before the opening parens of the tuple. Is that going to be alright?

File: Rubberduck.Parsing/Grammar/VBAParser.g4 (hunk 1 above)
Comment (retailcoder) [reply Rubberduck.Parsing/Grammar/VBAParser.g4]: Doesn't affect anything really though

Comment (MDoerner) [reply Rubberduck.Parsing/Grammar/VBAParser.g4]: That is just me forgetting from where I moved the other special forms when trying to resolve the SLL problem.

I will revert that.

**Comment by MDoerner** [comment]
It will. The LL parser parses everything you can define with an antlr grammar as long as there is no indirect left-recursion.
Moreover, it is the highest precedence rule that can match the input.

**Comment by mansellan** [comment]
Just checking - `LINE` isn't in that list, is that handled elsewhere or out of scope?

**Pull request merged**